packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -64,15 +94,103 @@ export default class Sayt extends LitElement {
     this.visible = true;
   }
 
+  /**
+   * Makes SAYT visible if the event refers to the correct SAYT component.
+   *
+   * @param event An event that can contain a searchbar ID.
+   */
+  showCorrectSayt(event: CustomEvent) {
+    if (this.isCorrectSayt(event)) {
+      this.showSayt();
+    }
+  }
+
   /**
    * Changes the `visible` property to `false`.
    */
   hideSayt() {
     this.visible = false;
   }
 
+  /**
+   * Hides SAYT if the event refers to the correct SAYT component.
+   *
+   * @param event An event that can contain a searchbar ID.
+   */
+  hideCorrectSayt(event: CustomEvent) {
+    if (this.isCorrectSayt(event)) {
+      this.hideSayt();
+    }
+  }
+
+  /**
+   * Determines whether an event refers to the correct SAYT. This is true if
+   * a matching `searchbar` ID is specified, if the event has no `searchbar`
+   * ID specified, or if SAYT has no `searchbar` ID specified.
+   *
+   * @param event An event that can contain a searchbar ID for comparison.
+   */
+  isCorrectSayt(event: CustomEvent) {
+    const searchbar = event.detail && event.detail.searchbar;
+    return this.searchbar === undefined ||
+      searchbar === this.searchbar ||
+      searchbar === undefined;
+  }
+
+  /**
+   * Processes a click event in order to close SAYT under the right conditions.
+   * 
+   * @param event The click event.
+   */
+  processClick(event: Event) {
+    const target = event.target as Node;
```

Comment:
I think this is here because I don't seem to have access to a `ClickEvent`, despite the fact that I have access to a `KeyboardEvent`. I get compile errors.

Reply:
I'll change to `MouseEvent`, but I'm still getting errors without casting.